polyval/src/field.rs

```diff
@@ -27,29 +40,64 @@ pub type Block = [u8; FIELD_SIZE];
 
 /// POLYVAL field element.
 #[derive(Copy, Clone)]
-pub struct Element<B: Backend>(B);
+pub enum Element {
+    #[cfg(all(
+        target_feature = "pclmulqdq",
+        target_feature = "sse2",
+        target_feature = "sse4.1",
+        any(target_arch = "x86", target_arch = "x86_64")
+    ))]
+    /// (P)CLMUL(QDQ)-accelerated backend on supported x86 architectures
+    Clmul(M128i),
 
-impl<B: Backend> Element<B> {
+    /// Portable software fallback
+    Soft(U64x2),
+}
+
+impl Element {
     /// Load a `FieldElement` from its bytestring representation.
+    #[cfg(all(
+        target_feature = "pclmulqdq",
+        target_feature = "sse2",
+        target_feature = "sse4.1",
+        any(target_arch = "x86", target_arch = "x86_64")
+    ))]
     pub fn from_bytes(bytes: Block) -> Self {
-        Element(bytes.into())
+        if cfg!(feature = "std") {
+            if is_x86_feature_detected!("pclmulqdq") {
```

Comment:
Note that it will not work on `no_std` targets, so we will have to use RDRAND directly (and cache results).